contracts/escrow_fee_distributor/tests/integration.rs

```diff
@@ -0,0 +1,1233 @@
+
+    assert_eq!(
+        vec![
+            attr("action", "update_config"),
+            attr("checkpoint_token_enabled", "true"),
+            attr("max_limit_accounts_of_claim", "20"),
+        ],
+        vec![
+            resp.events[1].attributes[1].clone(),
+            resp.events[1].attributes[2].clone(),
+            resp.events[1].attributes[3].clone(),
+        ]
+    );
+}
+
+#[test]
+fn claim_without_fee_on_distributor() {
+    let mut router = mock_app();
+    let router_ref = &mut router;
+    let owner = Addr::unchecked(OWNER.clone());
+    let user1 = Addr::unchecked(USER1.clone());
+    let user2 = Addr::unchecked(USER2.clone());
+
+    let base_pack = init_astroport_test_package(router_ref).unwrap();
+
+    let xastro_token = base_pack.get_staking_xastro(router_ref);
+
+    // sets 200_000_000 xASTRO tokens to user1
+    mint(
+        router_ref,
+        base_pack.staking.clone().unwrap().address,
+        xastro_token.clone(),
+        &user1,
+        200,
+    );
+
+    // sets 200_000_000 xASTRO tokens to user2
+    mint(
+        router_ref,
+        base_pack.staking.clone().unwrap().address,
+        xastro_token.clone(),
+        &user2,
+        200,
+    );
+
+    // locks 100 xASTRO from user1 for WEEK * 2
+    base_pack
+        .create_lock(router_ref, user1.clone(), WEEK * 104, 200)
+        .unwrap();
+
+    // locks 100 xASTRO from user2 for WEEK * 2
+    base_pack
+        .create_lock(router_ref, user2.clone(), WEEK * 104, 200)
+        .unwrap();
+
+    // going to the last week
+    router_ref.update_block(next_block);
+    router_ref.update_block(|b| b.time = b.time.plus_seconds(WEEK * 103));
+
+    // set checkpoints for each week
+    router_ref
+        .execute_contract(
+            owner.clone(),
+            base_pack.escrow_fee_distributor.clone().unwrap().address,
+            &ExecuteMsg::CheckpointToken {},
+            &[],
+        )
+        .unwrap();
+
+    // check if tokens for last week equal to 0
+    let resp: Vec<Uint128> = router_ref
+        .wrap()
+        .query_wasm_smart(
+            &base_pack.escrow_fee_distributor.clone().unwrap().address,
+            &QueryMsg::FeeTokensPerWeek {
+                start_after: Some(get_period(router_ref.block_info().time.seconds() - WEEK)),
+                limit: None,
+            },
+        )
+        .unwrap();
+    assert_eq!(vec![Uint128::new(0)], resp);
+
+    // check if voting supply per week is set
+    let resp: Vec<Uint128> = router_ref
+        .wrap()
+        .query_wasm_smart(
+            &base_pack.escrow_fee_distributor.clone().unwrap().address,
+            &QueryMsg::VotingSupplyPerWeek {
+                start_after: None,
+                limit: Some(2),
+            },
+        )
+        .unwrap();
+    assert_eq!(
+        vec![Uint128::new(1000000000), Uint128::new(990384615)],
+        resp
+    );
+
+    // try to claim fee for user1
+    router_ref
```

Comment:
Also check that users actually received rewards.

Reply:
Why we should check this for a distributor?